lib/utils/compose_ts.ts

```diff
@@ -53,10 +53,20 @@ import Logger = require('./logger');
  * @param buildargs --buildArg flag input in dictionary format
  */
 async function parseMetadataFromFlags(
+	dir: string,
 	buildargs: Dictionary<string>,
 	composition?: Composition,
 ): Promise<ParsedBuildArguments> {
-	const serviceNames = Object.keys(composition?.services || {});
+	const yaml = await import('js-yaml');
+	const mergedComposition: Composition = yaml.load(
```

Comment:
The build was failing on this line with the error:
```
lib/utils/compose_ts.ts(61,8): error TS2322: Type 'string | number | object | null | undefined' is not assignable to type 'Composition'.
  Type 'undefined' is not assignable to type 'Composition'.
```
I've added a commit that fixes it, and also refactors the code to avoid the `yaml.load` and `yaml.dump` in this function (I was a bit worried about `yaml.dump` because of #2226 and #2227, although I think it would be fine here).